payjoin-directory/src/config.rs

```diff
@@ -35,18 +56,48 @@ impl Config {
             timeout: Duration::from_secs(built_config.get("timeout")?),
             storage_dir: built_config.get("storage_dir")?,
             ohttp_keys: built_config.get("ohttp_keys")?,
+            #[cfg(feature = "acme")]
+            acme: if built_config.get_table("acme").is_ok() {
+                Some(AcmeConfig {
+                    domain: built_config.get("acme.domain")?,
+                    contact: built_config.get("acme.contact")?,
+                    lets_encrypt_production: built_config.get("acme.lets_encrypt_production")?,
+                    cache_dir: built_config.get("acme.cache_dir")?,
+                })
+            } else {
+                None
+            },
         })
     }
 }
 
 fn add_defaults(config: Builder, cli: &Cli) -> Result<Builder, ConfigError> {
-    config
+    let config = config
         .set_override_option("listen_addr", Some(format!("[::]:{}", cli.port)))?
         .set_override_option(
             "metrics_listen_addr",
             Some(format!("localhost:{}", cli.metrics_port)),
         )?
         .set_override_option("timeout", Some(cli.timeout))?
         .set_override_option("ohttp_keys", Some(cli.ohttp_keys.to_string_lossy().into_owned()))?
-        .set_override_option("storage_dir", Some(cli.storage_dir.to_string_lossy().into_owned()))
+        .set_override_option("storage_dir", Some(cli.storage_dir.to_string_lossy().into_owned()))?;
+
+    #[cfg(feature = "acme")]
+    let config = if cli.acme.domain.is_some() {
+        config
+            .set_override_option("acme.domain", cli.acme.domain.clone())?
+            .set_override_option("acme.contact", cli.acme.contact.clone())?
+            .set_override_option(
+                "acme.lets_encrypt_production",
+                Some(!cli.acme.lets_encrypt_staging),
+            )?
+            .set_override_option(
+                "acme.cache_dir",
+                cli.acme.cache_dir.clone().map(|s| s.to_string_lossy().into_owned()),
+            )?
+    } else {
+        config
+    };
```

Comment:
I don't think you need the else. config is already defined on 75, unless you did this explicitly to prevent depending on that let config = config ... above?